src/qt3utils/datagenerators/spectrometers/princeston.py

```diff
@@ -152,67 +159,45 @@ def close(self) -> None:
         """
         self._automation.Dispose()
         logger.info('Closed AddInProcess.exe')
-        
+
     def __del__(self) -> None:
         """
         Uses Python garbage collection method used to terminate
         AddInProcess.exe, if this is not done, LightField will not reopen
         properly.
         """
         self.close()
-        logger.info('Closed AddInProcess.exe')
 
-class SpectrometerConfig():
 
-    """ 
-    This constant represents the minimum difference your maximum and minimum wavelength have to be away from each other in order to perform 'acquire_step_and_glue".
-    If the difference between lambda_max and lambda_min is less than this you will run into an error.
-    """
+class PrincetonSpectrometer(Spectrometer):
+
+    DEVICE_NAME: str = 'Princeton Spectrometer'
+    ACQUISITION_MODES: set[str] = {'single', 'step-and-glue'}
+
     MIN_WAVELENGTH_DIFFERENCE = 117
-    
-    def __init__(self, experiment_name: str = None) -> None:
+    """
+    This constant represents the minimum difference your maximum 
+    and minimum wavelength have to be away from each other in 
+    order to perform 'acquire_step_and_glue". If the difference between 
+    lambda_max and lambda_min is less than this you will run into an error.
+    """
+
+    def __init__(self, experiment_name: str = None):
+        super().__init__()
         self._experiment_name = experiment_name
         self.light = LightfieldApplicationManager(True)
 
-    def finalize(self) -> None:
+    def open(self) -> None:
+        """ Opens the Lifhtfield application. """
```

Comment:
*Lightfield. Also please provide me with a situation when this would be used. Is this just basically a bigger version of a flag?

Reply:
This will be used in the case where you open the lightfield application and accidentally try to reopen it. This will hopefully prevent any issues with duplicate application

Reply:
Good catch